pages/home/oai-agents/user-auth-interrupts.mdx

```diff
@@ -0,0 +1,194 @@
+github_agent = Agent(
+    name="GitHub agent",
+    instructions="You are a helpful assistant that can assist with GitHub API calls.",
+    model="gpt-4o-mini",
+    tools=tools,
+)
+```
+
+### Handle authorization errors
+
+When a user needs to authorize access to a tool, the agent will raise an `AuthorizationError`. You can handle it like this:
+
+```python
+try:
+    result = await Runner.run(
+        starting_agent=github_agent,
+        input="Star the arcadeai/arcade-ai repo",
+        # Pass a unique user_id for authentication
+        context={"user_id": "user@example.com"},
+    )
+    print("Final output:\n\n", result.final_output)
+except AuthorizationError as e:
+    # Display the authorization URL to the user
+    print(f"Please Login to GitHub: {e}")
+    # The error contains the authorization URL that the user should visit
+```
+
+### Wait for authorization completion
+
+You can also wait for the user to complete the authorization before continuing:
+
+```python
+from arcadepy.auth import wait_for_auth, is_authorized
+import asyncio
+
+async def handle_auth_flow(auth_id):
+    # Display a message to the user
+    print("Please visit the authorization URL in your browser")
+
+    # Wait for the user to complete authorization
+    await wait_for_auth(auth_id)
+
+    # Check if authorization was successful
+    if await is_authorized(auth_id):
+        print("Authorization successful! You can now use the tool.")
+        return True
+    else:
+        print("Authorization failed or timed out.")
+        return False
+
+# In your main function
+try:
+    # Run agent code
+    # ...
+except AuthorizationError as e:
+    auth_id = e.auth_id
+    if await handle_auth_flow(auth_id):
+        # Try running the agent again
+        result = await Runner.run(
+            starting_agent=github_agent,
+            input="Star the arcadeai/arcade-ai repo",
+            context={"user_id": "user@example.com"},
+        )
+        print("Final output:\n\n", result.final_output)
+```
+
+### Complete example
+
+Here's a full example that demonstrates the authorization flow:
+
+```python
+from agents import Agent, Runner
+from arcadepy import AsyncArcade
+from agents_arcade import get_arcade_tools
+from agents_arcade.errors import AuthorizationError
+
+async def main():
+    client = AsyncArcade()
+    # Use the "github" toolkit for this example
+    tools = await get_arcade_tools(client, ["github"])
+
+    github_agent = Agent(
+        name="GitHub agent",
+        instructions="You are a helpful assistant that can assist with GitHub API calls.",
+        model="gpt-4o-mini",
+        tools=tools,
+    )
+
+    try:
+        result = await Runner.run(
+            starting_agent=github_agent,
+            input="Star the arcadeai/arcade-ai repo",
+            # make sure you pass a UNIQUE user_id for auth
+            context={"user_id": "user@example.com"},
+        )
+        print("Final output:\n\n", result.final_output)
```

Comment:
I think this example should show the `wait_for_auth` pattern, since we're describing it as a complete example.